libraries/EEPROM/EEPROM.h

```diff
@@ -53,8 +53,11 @@ class EEPROMClass {
     if (address < 0 || address + sizeof(T) > _size)
       return t;
 
-    memcpy(_data + address, (const uint8_t*) &t, sizeof(T));
-    _dirty = true;
+    _dirty = (memcmp(_data + address, (const uint8_t*) &t, sizeof(T)) !=0);
```

Comment:
I could modify the code in this way:
    if (memcmp(_data + address, (const uint8_t*) &t, sizeof(T)) !=0) {
         _dirty = true;
	  memcpy(_data + address, (const uint8_t*) &t, sizeof(T));
   }
It will grant that the _dirty flag is reset only at EEPROM.commit() and no changes are lost